src/test/java/at/ac/tuwien/kr/alpha/grounder/SubstitutionTest.java

```diff
@@ -79,6 +80,38 @@ public void unifyTermsFunctionTermBinding() throws Exception {
 		assertEquals(A, substitution.eval(Y));
 	}
 
+	@Test
+	public void extendSubstitution() {
+		VariableTerm varX = VariableTerm.getInstance("X");
+		VariableTerm varY = VariableTerm.getInstance("Y");
+		Substitution sub1 = new Substitution();
+		sub1.put(varX, varY);
+		Substitution sub2 = new Substitution();
+		sub2.put(varY, ConstantTerm.getInstance("a"));
+
+		sub1.extendWith(sub2);
+		BasicAtom atom1 = parseAtom("p(X)");
+
+		Atom atomSubstituted = atom1.substitute(sub1);
+		assertEquals(ConstantTerm.getInstance("a"), atomSubstituted.getTerms().get(0));
+	}
+
+	@Test
+	public void mergeSubstitutionsIntoLeft() {
+		VariableTerm varX = VariableTerm.getInstance("X");
+		VariableTerm varY = VariableTerm.getInstance("Y");
+		VariableTerm varZ = VariableTerm.getInstance("Z");
+		Term constA = ConstantTerm.getInstance("a");
+		Substitution left = new Substitution();
+		left.put(varX, varY);
+		left.put(varZ, varY);
+		Substitution right = new Substitution();
+		right.put(varX, constA);
+		Substitution merged = Substitution.mergeIntoLeft(left, right);
+		assertEquals(merged.eval(varY), constA);
```

Comment:
The first parameter of `assertEquals` is `expected`, the second one is `actual`.